icu4j/main/classes/core/src/com/ibm/icu/impl/LocaleFallbackData.java

```diff
@@ -0,0 +1,562 @@
+        { "sat", "Olck" },
+        { "saz", "Saur" },
+        { "sck", "Deva" },
+        { "scl", "Arab" },
+        { "sd", "Arab" },
+        { "sd_IN", "Deva" },
+        { "sdh", "Arab" },
+        { "sga", "Ogam" },
+        { "sgw", "Ethi" },
+        { "shi", "Tfng" },
+        { "shn", "Mymr" },
+        { "shu", "Arab" },
+        { "si", "Sinh" },
+        { "skr", "Arab" },
+        { "smp", "Samr" },
+        { "sog", "Sogd" },
+        { "sou", "Thai" },
+        { "sr", "Cyrl" },
+        { "srb", "Sora" },
+        { "srx", "Deva" },
+        { "swb", "Arab" },
+        { "swv", "Deva" },
+        { "syl", "Beng" },
+        { "syr", "Syrc" },
+        { "ta", "Taml" },
+        { "taj", "Deva" },
+        { "tcy", "Knda" },
+        { "tdd", "Tale" },
+        { "tdg", "Deva" },
+        { "tdh", "Deva" },
+        { "te", "Telu" },
+        { "tg", "Cyrl" },
+        { "tg_PK", "Arab" },
+        { "th", "Thai" },
+        { "thl", "Deva" },
+        { "thq", "Deva" },
+        { "thr", "Deva" },
+        { "ti", "Ethi" },
+        { "tig", "Ethi" },
+        { "tkt", "Deva" },
+        { "trw", "Arab" },
+        { "tsd", "Grek" },
+        { "tsf", "Deva" },
+        { "tsj", "Tibt" },
+        { "tt", "Cyrl" },
+        { "tts", "Thai" },
+        { "txg", "Tang" },
+        { "txo", "Toto" },
+        { "tyv", "Cyrl" },
+        { "udi", "Aghb" },
+        { "udm", "Cyrl" },
+        { "ug", "Arab" },
+        { "ug_KZ", "Cyrl" },
+        { "ug_MN", "Cyrl" },
+        { "uga", "Ugar" },
+        { "uk", "Cyrl" },
+        { "unr", "Beng" },
+        { "unr_NP", "Deva" },
+        { "unx", "Beng" },
+        { "ur", "Arab" },
+        { "uz_AF", "Arab" },
+        { "uz_CN", "Cyrl" },
+        { "vai", "Vaii" },
+        { "wal", "Ethi" },
+        { "wbq", "Telu" },
+        { "wbr", "Deva" },
+        { "wni", "Arab" },
+        { "wsg", "Gong" },
+        { "wtm", "Deva" },
+        { "wuu", "Hans" },
+        { "xco", "Chrs" },
+        { "xcr", "Cari" },
+        { "xlc", "Lyci" },
+        { "xld", "Lydi" },
+        { "xmf", "Geor" },
+        { "xmn", "Mani" },
+        { "xmr", "Merc" },
+        { "xna", "Narb" },
+        { "xnr", "Deva" },
+        { "xpr", "Prti" },
+        { "xsa", "Sarb" },
+        { "xsr", "Deva" },
+        { "yi", "Hebr" },
+        { "yue", "Hant" },
+        { "yue_CN", "Hans" },
+        { "zdj", "Arab" },
+        { "zgh", "Tfng" },
+        { "zh", "Hans" },
+        { "zh_AU", "Hant" },
+        { "zh_BN", "Hant" },
+        { "zh_GB", "Hant" },
+        { "zh_GF", "Hant" },
+        { "zh_HK", "Hant" },
+        { "zh_ID", "Hant" },
+        { "zh_MO", "Hant" },
+        { "zh_PA", "Hant" },
+        { "zh_PF", "Hant" },
+        { "zh_PH", "Hant" },
+        { "zh_SR", "Hant" },
+        { "zh_TH", "Hant" },
+        { "zh_TW", "Hant" },
+        { "zh_US", "Hant" },
+        { "zh_VN", "Hant" },
+        { "zhx", "Nshu" },
+        { "zkt", "Kits" },
+    };
```

Comment:
About the string pairs. When Java has an array, the initialization code builds it item by item. If you are going to want a map anyway, and you are generating the code, then it is just as easy to avoid the extra work and generate the code to build the map in the first place. I'll phrase this as using an ImmutableMap builder.

OLD
    public static final String[][] DEFAULT_SCRIPT_TABLE = {
            { "ab", "Cyrl" },
...
            { "zkt", "Kits" },
    };

NEW
    public static final Map<String, String> DEFAULT_SCRIPT_MAP = ImmutableMap.<String, String>builder()
            .put("ab", "Cyrl")
...
            .put("zkt", "Kits")
            .build();

If ImmutableMap isn't available to you, then you can add a simple helper function:

    public static class ImmutableMapBuilder<K,V> {
        private Map<K,V> data;
        public ImmutableMapBuilder(Map<K,V> data) {
            this.data = data;
        }
        public ImmutableMapBuilder<K,V> put(K key, V value) {
            data.put(key, value);
            return this;
        }
        public Map<K, V> build() {
            return Collections.unmodifiableMap(data);
        }
    }

and change 
ImmutableMap.<String, String>builder() 
to 
new ImmutableMapBuilder<String,String>(new HashMap<>())

The other advantage is that an array in Java is mutable, so you really never want to expose it as public; that's not a problem if you just directly build an immutable map.

Reply:
You know, I like that a lot better.  I went ahead and made this change, although since I don't have ImmutableMap, instead of creating a helper class like you show above (which _is_ nice), I just did the builder code the old-fashioned way.